src/python_extension.ts

```diff
@@ -84,28 +111,33 @@ export class PythonExtension {
 
 
 	addEnvSelector():void {
-		log("add virtual env selector");
 		// Add the EnvSelectorWindow
 		this.inmantaEnvSelector = window.createStatusBarItem(StatusBarAlignment.Right);
 		this.inmantaEnvSelector.command = "python.setInterpreter";
 		this.inmantaEnvSelector.tooltip = "Select a virtual environment";
 		// Update the button visibility when the extension is activated
 		this.updateInmantaEnvVisibility();
-		// Update the button visibility when the active editor changes
-		window.onDidChangeActiveTextEditor(()=>this.updateInmantaEnvVisibility());
 		this.registerCallbackOnChange(()=>this.updateInmantaEnvVisibility());
 
 	}
 
 
 	/**
 	 * register a function that will be called when the "onChange" function is called
-	 * @param {() => void} onChangeCallback A function
+	 * @param {(newPath?) => void} onChangeCallback A function
 	 */
-	registerCallbackOnChange(onChangeCallback: () => void) {
+	registerCallbackOnChange(onChangeCallback: (newPath?, outermost?) => void, tags?) {
```

Comment:
the typing should be done like this but I'm not sure why `tags` is needed? 
```suggestion
	registerCallbackOnChange(onChangeCallback: (newPath?, outermost?) => void | tags) {
```